packages/babel-generator/src/generators/types.js

```diff
@@ -120,8 +120,11 @@ export function NullLiteral() {
 
 export function NumericLiteral(node: Object) {
   const raw = this.getPossibleRaw(node);
+  const opts = this.format.jsescOption;
   const value = node.value + "";
-  if (raw == null) {
+  if (this.format.jsescOption.numbers) {
+    this.number(jsesc(parseInt(value), opts));
```

Comment:
Why use `parseInt` when we already have the original value at `node.value`?

Reply:
`jsesc` was not working with `raw` until I added `parseInt` to it. Thinking the same, I added it to `value` too. 😅